atc/worker/worker.go

```diff
@@ -658,7 +658,7 @@ func (worker *gardenWorker) cloneRemoteVolumes(
 			},
 			container,
 			teamID,
-			"",
+			filepath.Join("/streamed", nonLocalInput.desiredMountPath),
```

Comment:
One problem with this is that `/streamed/desired/mount/path` *could* be a valid `desiredMountPath` for another input. Granted, it's unlikely to come up, but still a potential bug. I think my original suggestion of `streamed:/desired/mount/path` technically couldn't come up (at least with the current state of the code), but for a non-obvious reason (somewhere higher up, we `filepath.Join` the input path to the working directory, which is always an absolute path).

Either way, it's all kind of a hack. I guess one other way would be for there to be a separate `streamed: bool` column in `volumes` and `FindOrCreateVolumeForContainer` takes the `path` and whether or not it's streamed, and when we try to find the volume, we match on both fields. This is effectively doing the same thing as just prefixing the path with `/streamed/...` or `streamed:...`, but is a bit more explicit.

WDYT? cc @clarafu

Reply:
@aoldershaw I don't feel this would be a potential bug, because Concourse doesn't support absolution task input path, thus `/streamed/...` shouldn't be conflict to any input paths. Even if that's really an issue, we can workaround with a UUID `/streamed/<uuid>` as mount path, that won't be conflict to any path.

Reply:
Hmm to be honest I'm fine with either, the advantages I could see with having a streamed boolean would be that it would allow us to easily query for streamed volumes vs non streamed volumes. Although, I can't really think of a need for that now other than to show the user in a `fly volumes` command.